_ext/eclipse-base/src/main/java/com/diffplug/spotless/extra/eclipse/base/osgi/ResourceAccessor.java

```diff
@@ -74,7 +74,7 @@ class ResourceAccessor {
 
 	private static BundleFile getBundlFile(Class<?> clazz) throws BundleException {
 		URI objUri = getBundleUri(clazz);
-		File jarOrDirectory = new File(objUri.getPath());
+		var jarOrDirectory = new File(objUri.getPath());
```

Comment:
Why wasn't the `URI` above replaced?

Reply:
At first sight, it would be a limitation around type resolution, due to file being processed one a per-file basis. Let me have a deeper look.

Reply:
`UnsolvedSymbolException{context='null', name='BundleException', cause='null'}`. One may argue this information is not relevant in this purpose. However, here it is: [JavaParser](https://github.com/javaparser/javaparser), the source -> AST used to manipulate Java-Code failed resolving the type of `getBundleUri(clazz)` as given method refers to a type unknown in the context of given source file.

Similar technical challenge around https://github.com/diffplug/spotless/issues/1379#issuecomment-1406875964.

For now, Cleanthat focuses on linting on a per-file basis (for similar reason Spotless processes (for now) each file individually).

Reply:
Thanks for the explanation!

Reply:
I would think that any assignment could be a `var` and we don't need to know anything about its type. The only reason not to `var` is if it's a declaration without an assignment. Am I wrong?

Reply:
You can find the unitTests for this mutator in [LocalVariableTypeInferenceCases](https://github.com/solven-eu/cleanthat/blob/master/java/src/test/java/eu/solven/cleanthat/engine/java/refactorer/cases/do_not_format_me/LocalVariableTypeInferenceCases.java). There is multiple cases with `@UnmodifiedMethod`, expressing switching to a `var` would be invalid.

Even `List<?> i = new ArrayList<String>();` should not be switched to `var` as you may later assign a `CopyOnWriteArrayList` while `var` would have turned semantically from `List` to `ArrayList`